src/Utils/ProcessUtils.cpp

```diff
@@ -95,4 +101,122 @@ void ProcessUtils::cleanup() {
   m_thread = nullptr;
 }
 
+int ExecuteAndMonitorSystemCommand(const std::string& command,
```

Comment:
this is copy/paste code

Reply:
I am trying to access this function `ExecuteAndMonitorSystemCommand` from Configuration subsystem. This method lives inside the compiler class as protected method.. and because of this reason, I refactor it out to ProcessUtils for now. This is one way of doing it.

I am also exploring another method suggested by @chungshien to use callback function mechanism. I am still not getting it to work yet.

Reply:
sure. But copy/paste code isn't a right way.

Reply:
I don't mean to copy/paste. I mean, my goal is to refactor `ExecuteAndMonitorSystemCommand` from compiler class and put it into ProcessUtils class.

Reply:
Sure. But in this case it should be removed from Compiler.